python/cudf/cudf/io/csv.py

```diff
@@ -61,9 +61,17 @@ def read_csv(
         path_or_data=filepath_or_buffer,
         compression=compression,
         iotypes=(BytesIO, StringIO),
+        byte_ranges=[byte_range] if byte_range else None,
+        clip_dummy_buffer=True if byte_range else False,
         **kwargs,
     )
 
+    # Adjust byte_range for clipped dummy buffers
+    use_byte_range = byte_range
+    if byte_range and isinstance(filepath_or_buffer, BytesIO):
+        if byte_range[1] == filepath_or_buffer.getbuffer().nbytes:
+            use_byte_range = (0, byte_range[1])
```

Comment:
As discussed above, this is where we reset `byte_range` to a zero offset when we are using a local dummy buffer.